### DISCUSSION TITLE
New Camomile Examples with a basic framework for poly/mono synths.

### DISCUSSION THREAD

**Discussion by alfonso73** [pierreguillot/Camomile#235] [Ideas]
Hi!
Would be great to have a very simple (DSP-wise a simple sine osc synth with an ADSR would be enough) example incorporating a polyphonic framework with sustain pedal and pitch bend support.
And combining the smart [gui.link] abstraction with the abstraction explained here https://github.com/pierreguillot/Camomile/issues/121#issuecomment-404784189.
I think that having a framework based on various abstractions for poly and mono synth with all the relevant needed things, voice priority in mono and voice stealing in poly, sustain pedal support, pitch bend (and even an MPE based framwork) would be of great help for creating new DSP algos in PD and then exported as Camomile VST3.
I'm trying to do some work for proper polyphony and sustain pedal and hope to get some things done sooner or later.
thanks and all the best.
a.

**Comment by pierreguillot** [comment]
I fully agree with you about these examples for synthesizers. In my opinion, the best place to put the them would the section [How-to-create-new-plugins](https://github.com/pierreguillot/Camomile/wiki/How-to-create-new-plugins) on the wiki. And/or perhaps new specific section that would look more like a tutorial would also be great! The adavantage is that anybody can contribute to it, improve the text, add link to patches, change the images, fix the typo and so on.
